util/exec.go

```diff
@@ -21,16 +23,17 @@ type Command struct {
 }
 
 // CommandWithResult executes a command and returns its output, while printing the same output to the given logger.
-func CommandWithResult(logger *log.Logger, command *Command) (output string, err error) {
+func CommandWithResult(ctx context.Context, logger *log.Logger, command *Command) (output string, err error) {
 	logger.Printf("$ %s", strings.Join(command.Command, " "))
 
 	var writer bytesutil.SynchronizedBuffer
-	c := exec.Command(command.Command[0], command.Command[1:]...)
+	c := exec.CommandContext(ctx, command.Command[0], command.Command[1:]...)
 	if command.Directory != "" {
 		c.Dir = command.Directory
 	}
 	c.Stdout = io.MultiWriter(logger.Writer(), &writer)
 	c.Stderr = c.Stdout
+	c.WaitDelay = 3 * time.Second
```

Comment:
I assume this is for Ollama. Please always document something like this: extra options / settings. 3 is also a very curious value.

Reply:
@bauersimon still putting this in the commit. but not commenting, so please do a follow up PR commenting it.